From 0f9bc326a2aff14c05cd32fdc060e05f01af3b7f Mon Sep 17 00:00:00 2001
From: Ian Thomas <ianthomas23@gmail.com>
Date: Mon, 11 Sep 2023 10:51:29 +0100
Subject: [PATCH] Support pandas 2.1 (#1276)

* Remove pandas 2.1 pin

* Be more careful reading rows from dataframe when bundling

* Use pandas.testing.assert_series_equal
---
 .github/workflows/test.yaml        |  3 +++
 datashader/bundling.py             | 15 +++++++++++----
 datashader/tests/test_datatypes.py |  6 +++---
 setup.py                           |  3 ++-
 4 files changed, 19 insertions(+), 8 deletions(-)

Index: datashader-0.15.2/datashader/bundling.py
===================================================================
--- datashader-0.15.2.orig/datashader/bundling.py
+++ datashader-0.15.2/datashader/bundling.py
@@ -307,12 +307,13 @@ def _convert_graph_to_edge_segments(node
     df = df.sort_index()
     df = df.reset_index()
 
-    if params.include_edge_id:
+    include_edge_id = params.include_edge_id
+    if include_edge_id:
         df = df.rename(columns={'id': 'edge_id'})
 
     include_weight = params.weight and params.weight in edges
 
-    if params.include_edge_id:
+    if include_edge_id:
         if include_weight:
             segment_class = WeightedSegment
         else:
@@ -326,8 +327,14 @@ def _convert_graph_to_edge_segments(node
     df = df.filter(items=segment_class.get_merged_columns(params))
 
     edge_segments = []
-    for edge in df.values:
+    for tup in df.itertuples():
+        edge = (tup.src_x, tup.src_y, tup.dst_x, tup.dst_y)
+        if include_edge_id:
+            edge = (tup.edge_id,) + edge
+        if include_weight:
+            edge += (getattr(tup, params.weight),)
         edge_segments.append(segment_class.create_segment(edge))
+
     return edge_segments, segment_class
 
 
@@ -394,7 +401,7 @@ class connect_edges(param.ParameterizedF
         edges, segment_class = _convert_graph_to_edge_segments(nodes, edges, p)
         return _convert_edge_segments_to_dataframe(edges, segment_class, p)
 
-directly_connect_edges = connect_edges # For bockwards compatibility; deprecated
+directly_connect_edges = connect_edges # For backwards compatibility; deprecated
 
 
 def minmax_normalize(X, lower, upper):
Index: datashader-0.15.2/datashader/tests/test_datatypes.py
===================================================================
--- datashader-0.15.2.orig/datashader/tests/test_datatypes.py
+++ datashader-0.15.2/datashader/tests/test_datatypes.py
@@ -648,11 +648,11 @@ class TestRaggedGetitem(eb.BaseGetitemTe
 
         result = s.get([4, 6])
         expected = s.iloc[[2, 3]]
-        self.assert_series_equal(result, expected)
+        pd.testing.assert_series_equal(result, expected)
 
         result = s.get(slice(2))
         expected = s.iloc[[0, 1]]
-        self.assert_series_equal(result, expected)
+        pd.testing.assert_series_equal(result, expected)
 
         assert s.get(-1) is None
         assert s.get(s.index.max() + 1) is None
@@ -662,7 +662,7 @@ class TestRaggedGetitem(eb.BaseGetitemTe
 
         result = s.get(slice('b', 'd'))
         expected = s.iloc[[1, 2, 3]]
-        self.assert_series_equal(result, expected)
+        pd.testing.assert_series_equal(result, expected)
 
         result = s.get('Z')
         assert result is None
Index: datashader-0.15.2/setup.py
===================================================================
--- datashader-0.15.2.orig/setup.py
+++ datashader-0.15.2/setup.py
@@ -49,6 +49,7 @@ extras_require = {
         'pytest-cov',
         'rasterio',
         'rioxarray',
+        'scikit-image',
         'spatialpandas',
     ],
     'examples': examples,
